### PR TITLE
Added ability to specify agent socket

### PR DESCRIPTION
**Pull request by alongoldboim**
backport to 3-1-release [https://github.com/net-ssh/net-ssh/pull/376]
@mfazekas thanks for the help!

### CODE REVIEW

File: lib/net/ssh.rb [hunk 1]
```diff
@@ -70,7 +70,7 @@ module SSH
       :known_hosts, :global_known_hosts_file, :user_known_hosts_file, :host_key_alias,
       :host_name, :user, :properties, :passphrase, :keys_only, :max_pkt_size,
       :max_win_size, :send_env, :use_agent, :number_of_password_prompts,
-      :append_supported_algorithms, :non_interactive
+      :append_supported_algorithms, :non_interactive, :password_prompt, :agent_socket_factory
```
Comment (simon3z) [inline lib/net/ssh.rb]: @alongoldboim why are you adding `:password_prompt` ?

File: lib/net/ssh.rb [hunk 2]
```diff
@@ -193,6 +193,9 @@ module SSH
     # * :non_interactive => non interactive applications should set it to true
     #   to prefer failing a password/etc auth methods vs asking for password
     #
+    # * :agent_socket_factory => enables the user to pass a lambda/block that will serve as the socket factory
```
Comment (simon3z) [inline lib/net/ssh.rb]: @alongoldboim I think that the extra white line is misplaced now.

Comment (alongoldboim) [inline lib/net/ssh.rb]: fixed.

File: lib/net/ssh.rb (hunk 1 above)
Comment (alongoldboim) [inline lib/net/ssh.rb]: correct, removed.

**Comment by simon3z** [comment]
LGTM :+1:  thanks @alongoldboim

**Comment by alongoldboim** [comment]
@mfazekas any ETA on when we will have the new release? Thanks.

**Comment by mfazekas** [comment]
I try to have it this week, but no guarantee

**Comment by mfazekas** [comment]
https://rubygems.org/gems/net-ssh/versions/3.2.0.rc1

**Comment by alongoldboim** [comment]
@mfazekas This pr isn't in this release?

**Comment by mfazekas** [comment]
https://rubygems.org/gems/net-ssh/versions/3.2.0.rc2

**Comment by alongoldboim** [comment]
@mfazekas Thanks!

**Comment by alongoldboim** [comment]
@mfazekas Hey, can we expect a stable release anytime soon?

**Comment by mfazekas** [comment]
https://rubygems.org/gems/net-ssh/versions/3.2.0

**Pull request merged**